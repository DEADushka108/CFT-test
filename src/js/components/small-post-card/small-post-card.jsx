import React from 'react';
import PropTypes from 'prop-types';
import {redirectToRoute} from '../../store/redirect/redirect';
import {connect} from 'react-redux';
import {postDetails} from '../../types/post';
import {AppRoute} from '../../utils/const';
import {userDetails} from '../../types/user';
import {findItemById, getBody} from '../../utils/utils';
import {getUsersList} from '../../store/users/selectors';

const SmallPostCard = (props) => {
  const {postCard, usersList, redirect} = props;
  const {id, userId, title, body} = postCard;
  const userInfo = findItemById(userId, usersList);
  const {username} = userInfo;

  return <article className="journal__small-card small-card">
    <div className="small-card__user" >
      <img className="small-card__user-image" src="./img/content/no-user.png" alt={username}/>
      <p className="small-card__user-name" onClick={() => {
        redirect(`${AppRoute.USER}/${userId}`);
      }}>
        {username}
      </p>
      <button className="small-card__delete-button" type="button">
        <svg className="small-card__delete-icon">
          <use xlinkHref="#remove-item"></use>
        </svg>
        <span className="visually-hidden">Delete</span>
      </button>
    </div>
    <div className="small-card__wrapper" onClick={() => {
      redirect(`${AppRoute.POSTS}/${id}`);
    }}>
      <img className="small-card__content-image" src="./img/content/no-image.png" alt={title}/>
      <h3 className="small-card__title">{title}</h3>
      <p className="small-card__text">{getBody(body)}</p>
    </div>
  </article>;
};

SmallPostCard.propTypes = {
  postCard: postDetails,
  usersList: PropTypes.arrayOf(userDetails),
  redirect: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  usersList: getUsersList(state),
});

const mapDispatchToProps = (dispatch) => ({
  redirect(route) {
    dispatch(redirectToRoute(route));
  },
});

export {SmallPostCard};
export default connect(mapStateToProps, mapDispatchToProps)(SmallPostCard);

