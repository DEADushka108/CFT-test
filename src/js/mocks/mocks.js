export * from './const/const';
export * from './posts/posts';
export * from './store/store';
export * from './users/users';
export * from './comments/comments';
