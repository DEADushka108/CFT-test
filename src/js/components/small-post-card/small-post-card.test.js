import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {store, post, usersList, noop} from '../../mocks/mocks';
import {SmallPostCard} from './small-post-card';

it(`Should render SmallPostCard component correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <SmallPostCard
            postCard={post}
            usersList={usersList}
            redirect={noop}
            onDeleteClick={noop}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
