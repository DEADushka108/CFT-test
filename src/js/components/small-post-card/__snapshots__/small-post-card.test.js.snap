// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SmallPostCard component correctly 1`] = `
<article
  className="journal__small-card small-card"
>
  <div
    className="small-card__user"
  >
    <img
      alt="Bret"
      className="small-card__user-image"
      src="./img/content/no-user.png"
    />
    <p
      className="small-card__user-name"
      onClick={[Function]}
    >
      Bret
    </p>
    <button
      className="small-card__delete-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="small-card__delete-icon"
      >
        <use
          xlinkHref="#remove-item"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        Delete
      </span>
    </button>
  </div>
  <div
    className="small-card__wrapper"
    onClick={[Function]}
  >
    <img
      alt="sunt aut facere repellat provident occaecati excepturi optio reprehenderit"
      className="small-card__content-image"
      src="./img/content/no-image.png"
    />
    <h3
      className="small-card__title"
    >
      sunt aut facere repellat provident occaecati excepturi optio reprehenderit
    </h3>
    <p
      className="small-card__text"
    >
      quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas...
    </p>
  </div>
</article>
`;
